### PR TITLE
Fix history file usage in SQL CLI tool.

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
Signed-off-by: Yury-Fridlyand <yuryf@bitquilltech.com>

### Description
Changes include:
* Fix usage of `FileHistory`.
* Fix reading `history_file` and `log_file` parameters from config.
* Fix interpreting the default or custom user value.

Config location: `~/.config/opensearchsql-cli/config`
Default location for saving history and log files: `~/.config/opensearchsql-cli`
Inspired by: https://github.com/prompt-toolkit/python-prompt-toolkit/blob/master/examples/prompts/history/persistent-history.py

See team review at https://github.com/Bit-Quill/opensearch-project-sql/pull/155

### Test
1. Run SQL CLI in `venv` (see instructions below)
2. Run a query
3. Exit from SQL CLI
4. Open it back
5. Hit arrow-up keyboard button

### How to test
Ref: https://github.com/opensearch-project/sql/blob/2.x/sql-cli/development_guide.md#development-environment-set-up

Do once:
```
cd sql-cli
pip install virtualenv
virtualenv venv
pip install --editable .
```
Then every time you want to debug SQL CLI:
```
<checkout>
cd sql-cli
source ./venv/bin/activate
```
To exit from `venv`:
```
deactivate
```
You can run SQL CLI and switch branch. It will keep working with given timeout until closed.
 
### Issues Resolved
Query history didn't persist across SQL CLI runs.
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by joshuali925 — APPROVED** [review]

File: sql-cli/src/opensearch_sql_cli/opensearchsql_cli.py [hunk 1]
```diff
@@ -57,9 +61,21 @@ def __init__(self, clirc_file=None, always_use_pager=False, use_aws_authenticati
         self.multiline_continuation_char = config["main"]["multiline_continuation_char"]
         self.multi_line = config["main"].as_bool("multi_line")
         self.multiline_mode = config["main"].get("multi_line_mode", "src")
+        self.history_file = config["main"]["history_file"]
+        self.log_file = config["main"]["log_file"]
```
Comment (derek-ho) [inline sql-cli/src/opensearch_sql_cli/opensearchsql_cli.py:65]: where is this being used?

Comment (Yury-Fridlyand) [reply sql-cli/src/opensearch_sql_cli/opensearchsql_cli.py:65]: History file used for history, log file - for logging errors. For example, if config has incorrect path to the history file.

Comment (derek-ho) [reply sql-cli/src/opensearch_sql_cli/opensearchsql_cli.py:65]: Got it - do you need to pass this in anywhere like you do the history file?

Comment (Yury-Fridlyand) [reply sql-cli/src/opensearch_sql_cli/opensearchsql_cli.py]: No actually. Good point, thank you. `self.log_file` removed in f79e5a959.

**Review by derek-ho — APPROVED** [review]
LGTM

**Review by penghuo — APPROVED** [review]
Thanks.

**Pull request merged**